src/charm.py

```diff
@@ -450,20 +439,12 @@ def on_update_status(self, event):
 
     def on_cni_relation_changed(self, event):
         """Run CNI relation changed hook."""
-        if not self.stored.tigera_configured:
+        if not cast(bool, self.stored.tigera_configured):
             self.on_config_changed(event)
 
         self.cni_to_calico_enterprise(event)
         self.configure_cni_relation()
-        self.set_active_status()
 
-    def on_cni_relation_joined(self, event):
-        """Run CNI relation joined hook."""
-        if not self.stored.tigera_configured:
-            self.on_config_changed(event)
-
-        self.cni_to_calico_enterprise(event)
-        self.configure_cni_relation()
```

Comment:
ok, so we just ignore this until the relation_changed hook?  Maybe that works okay

Reply:
indeed -- we'll never have a useful `relation-joined` event in this charm.  it fires too early:

https://juju.is/docs/sdk/relation-name-relation-joined-event

the useful bits from the relation databag come during `relation-changed`, so no need for both / duplicate functions here.